File: engine/src/main/kotlin/com/stopptanz/engine/RandomSource.kt
package com.stopptanz.engine

fun interface RandomSource {
    fun nextLong(min: Long, max: Long): Long
}

class DefaultRandomSource(private val random: kotlin.random.Random = kotlin.random.Random.Default) : RandomSource {
    override fun nextLong(min: Long, max: Long): Long {
        if (min == max) return min
        return random.nextLong(min, max + 1)
    }
}

File: web/src/engine/randomSource.ts
// Copyright (C) 2026 Dehalion
// SPDX-License-Identifier: GPL-3.0-only
export type RandomSource = (min: number, max: number) => number

export const defaultRandomSource: RandomSource = (min, max) => {
  if (min === max) return min
  return min + Math.floor(Math.random() * (max - min + 1))
}
